lib/src/wallet_connect/session.dart

```diff
@@ -319,13 +322,17 @@ class WalletConnection extends ValueListenable<WalletConnectState> {
           break;
       }
     } catch (e) {
-      final response = JsonRpcResponse.internalError(jsonRequest.id);
-      final publishMessage =
-          PublishSinkMessage.publish(_remotePeerId!, response);
-      _outSink?.add(publishMessage);
+      _handleRequestErrors(e, jsonRequest.id);
```

Comment:
It is worth noting that a try/catch will not catch an error thrown inside of an async method call unless that async method is awaited.

The error will instead go to the zone handler, such as that from `runZonedGuarded`

Now that we added `onError`, the error will go there and also not be thrown in either the zone or this try/catch, so we should remove the try/catch since it is unused

Reply:
There was one method that wasn't an async one, that's why I left the try/catch.

Reply:
In that case, would be good to surround just that method with try/catch for clarity or change the method to async for uniformity